File: v2/tests/code-intelligence/parser/fixtures/sample.js
export function greet(name) {
  return `Hello, ${name}!`;
}

export class UserService {
  constructor() {
    this.users = new Map();
  }

  async getUser(id) {
    return this.users.get(id) || null;
  }

  createUser(name, email) {
    const user = { id: crypto.randomUUID(), name, email };
    this.users.set(user.id, user);
    return user;
  }
}

File: v2/tests/code-intelligence/parser/fixtures/sample.rs
pub fn greet(name: &str) -> String {
    format!("Hello, {}!", name)
}

pub struct UserService {
    users: HashMap<String, User>,
}

impl UserService {
    pub fn new() -> Self {
        Self { users: HashMap::new() }
    }

    pub async fn get_user(&self, id: &str) -> Option<&User> {
        self.users.get(id)
    }
}
